Add references and fix direct product

Akshobhya1234 opened the pull request:
Clearly differentiate product, direct product, and direct sum with definitions in the literature. 
Give reference to the current issue and state what is out of scope of the thesis.

Akshobhya1234 committed:
Add references and fix direct product

JacquesCarette reviewed (changes requested):
So the section is still titled 'Direct Product' and it refers to another chapter which still talks about 'direct product'. You should make the thesis self-consistent.

The definition of product from Sanella is the 'usual' one that most expect, i.e. Cartesian product.

Akshobhya1234 commented:
> So the section is still titled 'Direct Product' and it refers to another chapter which still talks about 'direct product'. You should make the thesis self-consistent.
> 
> The definition of product from Sanella is the 'usual' one that most expect, i.e. Cartesian product.

I call it 'Direct Product' since they are called as 'direct product' in the standard library. I give the definition of 'product' from Sanella and say that when we talk about 'direct product' we are dealing with product. The definition in Chapter 2 is the correct definition of 'Direct Product' from Sanella. 

The term 'Cartesian product' is usually used when dealing with sets and the term 'Product' is used for algebra. I also clarify this in the same section.

The same goes for 'Morphism'. I call the section 'morphism' since they are available under that module in the standard library.

Akshobhya1234 committed:
Merge branch 'master' of https://github.com/Akshobhya1234/MSc-thesis into direct_product

JacquesCarette commented:
You shouldn't take the standard library as "conveyor of truth", and indeed, you should criticize it for choosing an odd name here and there. You should take a standard textbook (like Sanella) as establishing standard vocabulary instead.

If you insist on keeping 'direct product' in there, then you need to say (early and often) that this choice is to be consistent with the naming in the standard library. And you should still criticize that non-standard choice!

'Cartesian product' occurs all over category theory, and is way way more general than in just set theory.

Akshobhya1234 committed:
Merge branch 'master' of https://github.com/Akshobhya1234/MSc-thesis into direct_product

Akshobhya1234 committed:
Fix direct product

Akshobhya1234 commented:
I added definitions of product, direct product, and direct sum in the universal algebra chapter. I changed the section to Product Algebra instead of Direct Product.

Akshobhya1234 committed:
Update UniversalAlgebra.tex

JacquesCarette commented on Agda.tex:
```diff
@@ -783,41 +783,30 @@ \section{Homomorphism In Agda}
+Chapter 2. A co-product of an algebra is the dual of the product algebra
+\cite{sannella2012foundations}. A bi-product of algebra is an algebra that is
+both a product and a co-product. The standard library defines objects that are
+products of appropriate algebras and addresses them as direct products. In many
```
addresses -> calls

Akshobhya1234 replied:
fixed

JacquesCarette commented on references.bib:
```diff
@@ -678,4 +678,18 @@ @book{judson2020abstract
+    url = {https://webspace.maths.qmul.ac.uk/r.a.bailey/MAS305/algnotes14.pdf},
+    year = {2023}
+  }
+  @misc{szabo2000linear,
```
Is this truly a 'misc' or is there a better way to refer to this work?

Akshobhya1234 replied:
It is a book. I fixed it.

Akshobhya1234 committed:
Merge branch 'master' of https://github.com/Akshobhya1234/MSc-thesis into direct_product

Akshobhya1234 committed:
final fixes
